charts/argo-cd/templates/argocd-applicationset/deployment.yaml

```diff
@@ -58,15 +58,31 @@ spec:
           image: {{ default .Values.global.image.repository .Values.applicationSet.image.repository }}:{{ default (include "argo-cd.defaultTag" .) .Values.applicationSet.image.tag }}
           imagePullPolicy: {{ default .Values.global.image.imagePullPolicy .Values.applicationSet.image.imagePullPolicy }}
           ports:
-            - name: http
-              containerPort: {{ (split ":" .Values.applicationSet.args.probeBindAddr)._1 }}
-              protocol: TCP
             - name: metrics
               containerPort: {{ (split ":" .Values.applicationSet.args.metricsAddr)._1 }}
               protocol: TCP
+            - name: probe
+              containerPort: {{ (split ":" .Values.applicationSet.args.probeBindAddr)._1 }}
+              protocol: TCP
             - name: webhook
               containerPort: 7000
               protocol: TCP
+          livenessProbe:
+            tcpSocket:
+              port: probe
+            initialDelaySeconds: {{ .Values.applicationSet.livenessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.applicationSet.livenessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.applicationSet.livenessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.applicationSet.livenessProbe.successThreshold }}
+            failureThreshold: {{ .Values.applicationSet.livenessProbe.failureThreshold }}
+          readinessProbe:
+            tcpSocket:
+              port: probe
+            initialDelaySeconds: {{ .Values.applicationSet.readinessProbe.initialDelaySeconds }}
+            periodSeconds: {{ .Values.applicationSet.readinessProbe.periodSeconds }}
+            timeoutSeconds: {{ .Values.applicationSet.readinessProbe.timeoutSeconds }}
+            successThreshold: {{ .Values.applicationSet.readinessProbe.successThreshold }}
+            failureThreshold: {{ .Values.applicationSet.readinessProbe.failureThreshold }}
```

Comment:
Not in upstream manifests

Reply:
how about making this optional same as in Dex?